src/main/java/SYSC4806/Controller/HomeController.java

```diff
@@ -1,11 +1,88 @@
+import SYSC4806.Repository.CourseRepository;
+import SYSC4806.Repository.LearningOutcomeRepository;
+import SYSC4806.Repository.ProgramRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
 
 @Controller
 public class HomeController {
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Autowired
+    private LearningOutcomeRepository learningOutcomeRepository;
+
+    @Autowired
+    private ProgramRepository programRepository;
+
     @GetMapping("/")
-    public String home() {
+    public String home(Model model) {
+        model.addAttribute("categories", categoryRepository.findAll());
+        model.addAttribute("courses", courseRepository.findAll());
+        model.addAttribute("learningoutcomes", learningOutcomeRepository.findAll());
+        model.addAttribute("programs", programRepository.findAll());
         return "index";
     }
+
+    //    Adding and Removing data for admin (milestone 2?)
+//    @PostMapping("/")
+//    public String addCategory(@ModelAttribute Category newCategory, Model model) {
+//        categoryRepository.save(newCategory);
+//        return this.home(model);
+//    }
+//    @PostMapping("/")
+//    public String removeCategory(@ModelAttribute Category oldCategory, Model model) {
+//        categoryRepository.delete(oldCategory);
+//        return this.home(model);
+//    }
+//
+//    @PostMapping("/")
+//    public String addCourse(@ModelAttribute Course newCourse, Model model) {
+//        courseRepository.save(newCourse);
+//        return this.home(model);
+//    }
+//    @PostMapping("/")
+//    public String removeCourse(@ModelAttribute Course oldCourse, Model model) {
+//        courseRepository.delete(oldCourse);
+//        return this.home(model);
+//    }
+//
+//    @PostMapping("/")
+//    public String addLearningOutcome(@ModelAttribute LearningOutcome newLearningOutcome, Model model) {
+//        learningOutcomeRepository.save(newLearningOutcome);
+//        return this.home(model);
+//    }
+//    @PostMapping("/")
+//    public String removeLearningOutcome(@ModelAttribute LearningOutcome oldLearningOutcome, Model model) {
+//        learningOutcomeRepository.delete(oldLearningOutcome);
+//        return this.home(model);
+//    }
+//
+//    @PostMapping("/")
+//    public String addProgram(@ModelAttribute Program newProgram, Model model) {
+//        programRepository.save(newProgram);
+//        return this.home(model);
+//    }
+//    @PostMapping("/")
+//    public String removeProgram(@ModelAttribute Program oldProgram, Model model) {
+//        programRepository.delete(oldProgram);
```

Comment:
Also just for future reference, it would be great if a separate branch was created for this (as in connecting the DB to the View) just so each feature can have their own branches as per the project requirements :)

Reply:
Sorry shouldve replied to all the comments earlier today! Thanks for all the tips n advice. Really helps me learn haha. Will fix everything tonight once i get kshamina's push!!!